sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -209,6 +209,18 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
       Row(2743272264L, 2180413220L))
   }
 
+  test("misc data_type function") {
+    val df = Seq(("a", false)).toDF("a", "b")
+
+    checkAnswer(
+      df.select(data_type($"a"), data_type($"b")),
```

Comment:
I can't think of a SQL db which queries the datatype without using the table schema. However, for example in MongoDB, you can get something like that using the $type operator or 'typeof'.